<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Andy Beck" />
  <title>Compare-Results</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="pandoc.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header>
<h1 class="title">Compare-Results</h1>
<p class="author">Andy Beck</p>
</header>
<h1 id="abstract">Abstract</h1>
<h1 id="introduction">Introduction</h1>
<p>The goal of this (mini) project is to compare the results from running the <a href="http://github.com/carjed/smaug-genetics">smaug-genetics</a> pipeline on the same dataset. Numerical differences were observed in some of the outputs, and this likely resulted from the github repository being put together after all the analyses had been run initially (likely multiple times per step and not necessarily in the order described in the repositoryâ€™s documentation).</p>
<p>Here in this analysis, we compare the results of the logistic regression models used to predict the mutation rates for each 7-mer mutation type, conditioned on 14 genomic features (for more details, see the paper <a href="https://www.nature.com/articles/s41467-018-05936-5">Extremely rare variants reveal patterns of germline mutation rate heterogeneity in humans</a>.</p>
<h1 id="data">Data</h1>
<p>Each of the six mutation types (AT_CG, AT_GC, AT_TA, GC_AT, GC_CG, GC_TA) has its own subdirectory in <code>output\logmod_data\coefs</code>. Inside each mutation type directory are 4096 files, each corresponding to one 7-mer with the mutation type at the center. Each of these files contains the following table (with no header row:)</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Covariate</th>
<th style="text-align: center;">Estimate</th>
<th style="text-align: center;">SE</th>
<th style="text-align: center;">Z</th>
<th style="text-align: center;">pval</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">(Intercept)</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="even">
<td style="text-align: center;">DP</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="odd">
<td style="text-align: center;">H3K4me1</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="even">
<td style="text-align: center;">H3K4me3</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="odd">
<td style="text-align: center;">H3K9ac</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="even">
<td style="text-align: center;">H3K9me3</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="odd">
<td style="text-align: center;">H3K27ac</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="even">
<td style="text-align: center;">H3K27me3</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="odd">
<td style="text-align: center;">H3K36me3</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="even">
<td style="text-align: center;">EXON</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="odd">
<td style="text-align: center;">CpGI</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="even">
<td style="text-align: center;">RR</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="odd">
<td style="text-align: center;">LAMIN</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="even">
<td style="text-align: center;">DHS</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="odd">
<td style="text-align: center;">TIME</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
<tr class="even">
<td style="text-align: center;">GC</td>
<td style="text-align: center;"><em>est</em></td>
<td style="text-align: center;"><em>SE</em></td>
<td style="text-align: center;"><em>Z</em></td>
<td style="text-align: center;"><em>p</em></td>
</tr>
</tbody>
</table>
<h1 id="method">Method</h1>
<h2 id="statistics-that-might-be-useful-to-capture">Statistics that might be useful to capture</h2>
<ol type="1">
<li>Difference in the beta values for each parameter</li>
<li>Does direction of statistic match?</li>
<li>Difference between pvalues</li>
</ol>
<h3 id="computig-the-above-statistics">Computig the above statistics</h3>
<p>For each mutation type, we run <code>src/compare_coef.py n</code> (n = 0, 1, 2, 3, 4, or 5), which generates a resulting file in <code>output/</code>. For each mutation type, we have the following table:</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Motif</th>
<th style="text-align: center;">Var</th>
<th style="text-align: center;">BetaDiff</th>
<th style="text-align: center;">SignBeta</th>
<th style="text-align: center;">PDiff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">AT_CG_AACACGC</td>
<td style="text-align: center;">(Intercept)</td>
<td style="text-align: center;">0.05011159999999926</td>
<td style="text-align: center;">True</td>
<td style="text-align: center;">5.348999999999999e-05</td>
</tr>
</tbody>
</table>
<p>Where SignBeta = true if the signs of the coefficients are the same</p>
</body>
</html>
