<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Andy Beck" />
  <title>Compare-Results</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
  </style>
  <link rel="stylesheet" href="pandoc.css" type="text/css" />
</head>
<body>
<div id="header">
<h1 class="title">Compare-Results</h1>
<h2 class="author">Andy Beck</h2>
</div>
<h1 id="abstract">Abstract</h1>
<h1 id="introduction">Introduction</h1>
<p>The goal of this (mini) project is to compare the results from running the <a href="http://github.com/carjed/smaug-genetics">smaug-genetics</a> pipeline on the same dataset. Numerical differences were observed in some of the outputs, and this likely resulted from the github repository being put together after all the analyses had been run initially (likely multiple times per step and not necessarily in the order described in the repository's documentation).</p>
<p>Here in this analysis, we compare the results of the logistic regression models used to predict the mutation rates for each 7-mer mutation type, conditioned on 14 genomic features (for more details, see the paper <a href="https://www.nature.com/articles/s41467-018-05936-5">Extremely rare variants reveal patterns of germline mutation rate heterogeneity in humans</a>.</p>
<h1 id="data">Data</h1>
<p>Each of the six mutation types (AT_CG, AT_GC, AT_TA, GC_AT, GC_CG, GC_TA) has its own subdirectory in <code>output\logmod_data\coefs</code>. Inside each mutation type directory are 4096 files, each corresponding to one 7-mer with the mutation type at the center. Each of these files contains the following table (with no header row:)</p>
<table>
<thead>
<tr class="header">
<th align="center">Covariate</th>
<th align="center">Estimate</th>
<th align="center">SE</th>
<th align="center">Z</th>
<th align="center">pval</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">(Intercept)</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="even">
<td align="center">DP</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="odd">
<td align="center">H3K4me1</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="even">
<td align="center">H3K4me3</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="odd">
<td align="center">H3K9ac</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="even">
<td align="center">H3K9me3</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="odd">
<td align="center">H3K27ac</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="even">
<td align="center">H3K27me3</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="odd">
<td align="center">H3K36me3</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="even">
<td align="center">EXON</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="odd">
<td align="center">CpGI</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="even">
<td align="center">RR</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="odd">
<td align="center">LAMIN</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="even">
<td align="center">DHS</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="odd">
<td align="center">TIME</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
<tr class="even">
<td align="center">GC</td>
<td align="center"><em>est</em></td>
<td align="center"><em>SE</em></td>
<td align="center"><em>Z</em></td>
<td align="center"><em>p</em></td>
</tr>
</tbody>
</table>
<h1 id="method">Method</h1>
<h2 id="statistics-that-might-be-useful-to-capture">Statistics that might be useful to capture</h2>
<ol style="list-style-type: decimal">
<li>Difference in the beta values for each parameter</li>
<li>Does direction of statistic match?</li>
<li>Difference between pvalues</li>
</ol>
<h3 id="computig-the-above-statistics">Computig the above statistics</h3>
<p>For each mutation type, we run <code>src/compare_coef.py n</code> (n = 0, 1, 2, 3, 4, or 5), which generates a resulting file in <code>output/</code>. For each mutation type, we have the following table:</p>
<table>
<thead>
<tr class="header">
<th align="center">Motif</th>
<th align="center">Var</th>
<th align="center">BetaDiff</th>
<th align="center">SignBeta</th>
<th align="center">PDiff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">AT_CG_AACACGC</td>
<td align="center">(Intercept)</td>
<td align="center">0.05011159999999926</td>
<td align="center">1</td>
<td align="center">5.348999999999999e-05</td>
</tr>
</tbody>
</table>
<p>Where SignBeta = true if the signs of the coefficients are the same</p>
<p>I then generated files for each covariate (and the intercept) using similar shell commands as the following:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">head</span> -1 AT_CG.csv <span class="kw">&gt;</span> Intercept.csv
<span class="kw">for</span> <span class="kw">x</span> in AT_CG.csv AT_GC.csv AT_TA.csv GC_AT.csv GC_CG.csv GC_TA.csv<span class="kw">;</span> <span class="kw">do</span> <span class="kw">cat</span> <span class="ot">$x</span> <span class="kw">|</span> <span class="kw">\</span>
<span class="kw">grep</span> <span class="st">&#39;(Intercept)&#39;</span> <span class="kw">&gt;&gt;</span> Intercept.csv<span class="kw">;</span> <span class="kw">done</span></code></pre></div>
<p>Or we can do it all in one script:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">#!/usr/bin/zsh</span>
<span class="kw">for</span> <span class="kw">y</span> in <span class="ot">$(</span><span class="kw">tail</span> -n +2 output/AT_CG.csv <span class="kw">|</span> <span class="kw">cut</span> -f2 -d <span class="st">&quot;,&quot;</span> <span class="kw">|</span> <span class="kw">sort</span> <span class="kw">|</span> <span class="kw">uniq</span><span class="ot">)</span><span class="kw">;</span> <span class="kw">do</span>
    <span class="kw">head</span> -1 output/AT_CG.csv <span class="kw">&gt;</span> output/feature_specific/$<span class="dt">{${y:gs/\(/}:gs/\)/}</span>.csv
    <span class="kw">for</span> <span class="kw">x</span> in output/[A,G]*<span class="kw">;</span> <span class="kw">do</span>
        <span class="kw">cat</span> <span class="ot">$x</span> <span class="kw">|</span> <span class="kw">grep</span> <span class="ot">$y</span> <span class="kw">&gt;&gt;</span> <span class="st">&quot;output/feature_specific/${</span><span class="ot">${y:gs</span><span class="er">/\(/</span><span class="ot">}</span><span class="st">:gs/\)/}.csv&quot;</span>
    <span class="kw">done</span>
<span class="kw">done</span></code></pre></div>
<p>For the most part, the differences between the betas appear to be distributed around 0, though there are some examples of large differences between the beta values.</p>
<h2 id="tracking-where-in-the-pipeline-differences-occured">Tracking where in the pipeline differences occured</h2>
<ol style="list-style-type: decimal">
<li>Acquisition and Processing of Reference Data
<ul>
<li>Human genome lengths <code>hg19.genome</code></li>
<li>Mine was missing a header, but this only would have mattered had the neg_bin model been run</li>
<li>GC content in 10kb windows <code>gc10kb.bed</code></li>
<li>Mine has more columns -&gt; compare the ones we have in common?</li>
<li>Fixed width windows <code>genome.{size}.sorted.bed</code></li>
<li><strong>exactly the same</strong></li>
</ul></li>
<li>VCF files -&gt; <strong>exactly the same</strong></li>
<li>singletons -&gt; <strong>exactly the same</strong></li>
</ol>
<h3 id="gc-content-in-10kb-windows">GC content in 10kb windows</h3>
<h4 id="grab-column-from-my-file">Grab column from my file</h4>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">cat</span> gc10kb.bed <span class="kw">|</span> <span class="kw">sed</span> -n <span class="st">&#39;1!p&#39;</span> <span class="kw">|</span>  <span class="kw">cut</span> -f2-4  <span class="kw">&gt;</span> ~/research/compare_results/gc10kb_andy.bed</code></pre></div>
<h4 id="grab-column-from-jeds">Grab column from Jed's</h4>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">cat</span> gc10kb.bed <span class="kw">|</span> <span class="kw">cut</span> -f2-4  <span class="kw">&gt;</span> ~/research/compare_results/gc10kb_jed.bed</code></pre></div>
<h4 id="compare-in-r">Compare in R</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(tidyverse)
andy &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;gc10kb_andy.bed&quot;</span>, <span class="dt">col_names=</span><span class="kw">c</span>(<span class="st">&quot;start&quot;</span>,<span class="st">&quot;end&quot;</span>,<span class="st">&quot;gc&quot;</span>))
jed &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;gc10kb_jed.bed&quot;</span>, <span class="dt">col_names=</span><span class="kw">c</span>(<span class="st">&quot;start&quot;</span>,<span class="st">&quot;end&quot;</span>,<span class="st">&quot;gc&quot;</span>))</code></pre></div>
<p><strong>Answer:</strong> my file contains %AT content in the column Jed's code assumes contains GC (D'oh!)</p>
<p>Models should be re-run, but only impact should be a change in sign of the beta values for gc content.</p>
<h3 id="step-by-step-through-the-analysis-again...">Step-by-step through the analysis (<em>again...</em>)</h3>
<ol style="list-style-type: decimal">
<li>Read and pre-process the data (getData from smaug library)</li>
</ol>
</body>
</html>
